src/pages/signin/LoginForm.js

```diff
@@ -65,165 +65,150 @@ const defaultProps = {
-                        onSubmitEditing={this.validateAndSubmitForm}
-                        autoCapitalize="none"
-                        autoCorrect={false}
-                        keyboardType={CONST.KEYBOARD_TYPE.EMAIL_ADDRESS}
-                        errorText={formErrorText}
-                        hasError={hasError}
-                    />
-                </View>
-                {!_.isEmpty(this.props.account.success) && <Text style={[styles.formSuccess]}>{this.props.account.success}</Text>}
-                {!_.isEmpty(this.props.closeAccount.success || this.props.account.message) && (
-                    // DotIndicatorMessage mostly expects onyxData errors, so we need to mock an object so that the messages looks similar to prop.account.errors
-                    <DotIndicatorMessage
-                        style={[styles.mv2]}
-                        type="success"
-                        messages={{0: this.props.closeAccount.success || this.props.account.message}}
-                    />
-                )}
-                {
-                    // We need to unmount the submit button when the component is not visible so that the Enter button
-                    // key handler gets unsubscribed and does not conflict with the Password Form
-                    this.props.isVisible && (
-                        <View style={[styles.mt5]}>
-                            <FormAlertWithSubmitButton
-                                buttonText={this.props.translate('common.continue')}
-                                isLoading={this.props.account.isLoading && this.props.account.loadingForm === CONST.FORMS.LOGIN_FORM}
-                                onSubmit={this.validateAndSubmitForm}
-                                message={serverErrorText}
-                                isAlertVisible={!_.isEmpty(serverErrorText)}
-                                containerStyles={[styles.mh0]}
-                            />
-                        </View>
-                    )
-                }
-            </>
-        );
-    }
+        Session.beginSignIn(parsedPhoneNumber.possible ? parsedPhoneNumber.number.e164 : loginTrim);
+    }, [login, props.closeAccount, props.network, setFormError]);
+
+    useEffect(() => {
+        Session.clearAccountMessages();
+    }, []);
+
+    useEffect(() => {
+        if (!canFocusInputOnScreenFocus() || !input.current || !props.isVisible) {
+            return;
+        }
+        input.current.focus();
+    }, [props.isVisible, input]);
+
+    useEffect(() => {
+        if (props.blurOnSubmit) {
+            input.current.blur();
+        }
+        if (!input.current || !props.isVisible) {
+            return;
+        }
+        input.current.focus();
+    }, [props.blurOnSubmit, props.isVisible, input]);
+
+    const formErrorText = formError ? props.translate(formError) : '';
+    const serverErrorText = ErrorUtils.getLatestErrorMessage(props.account);
```

Comment:
Maybe we can use `useMemo` to memoize and avoid rerendering the child components?

Reply:
@luacmartins 
Updated to use the useMemo.